de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+        val httpVerbs = <ServiceAspect>newArrayList
+        httpVerbs.add("GetMapping".createServiceAspect)
+        httpVerbs.add("PutMapping".createServiceAspect)
+        httpVerbs.add("PostMapping".createServiceAspect)
+        httpVerbs.add("DeleteMapping".createServiceAspect)
+        httpVerbs.add("OptionsMapping".createServiceAspect)
+        httpVerbs.add("HeadMapping".createServiceAspect)
+        httpVerbs.add("PatchMapping".createServiceAspect)
+        httpVerbs.add("TraceMapping".createServiceAspect)
+        httpVerbs.add("Unspecified".createServiceAspect)
+        myTechModel.serviceAspects.addAll(httpVerbs)
+    }
+
+    def ServiceAspect createServiceAspect(String name) {
+        val ret = TECHNOLOGY_FACTORY.createServiceAspect
+        ret.name = name
+        ret.joinPoints.add(JoinPointType.OPERATIONS)
+        ret.features.add(AspectFeature.SINGLE_VALUED)
+        return ret
+    }
+
+    /** Creates OpenApi data types corresponding to <italic>OpenApi v3.0.3 - Data Types</italic>.
+     * OpenApi data type formats are represented as individual types.
+     */
+    def void initPrimitives(Technology technology) {
+        val primTypes = <TechnologySpecificPrimitiveType>newArrayList
+
+        primTypes.add(createPrimitive("Integer", true, DATA_FACTORY.createPrimitiveInteger))
+        primTypes.add(createPrimitive("Int32", false, DATA_FACTORY.createPrimitiveInteger))
+        primTypes.add(createPrimitive("Int64", true, DATA_FACTORY.createPrimitiveLong))
+        primTypes.add(createPrimitive("Number", true, DATA_FACTORY.createPrimitiveFloat))
+        primTypes.add(createPrimitive("Float", false, DATA_FACTORY.createPrimitiveFloat))
+        primTypes.add(createPrimitive("Double", true, DATA_FACTORY.createPrimitiveDouble))
+        primTypes.add(createPrimitive("String", true, DATA_FACTORY.createPrimitiveString))
+        primTypes.add(createPrimitive("Byte", true, DATA_FACTORY.createPrimitiveByte))
+        primTypes.add(createPrimitive("Binary", false, DATA_FACTORY.createPrimitiveString))
+        primTypes.add(createPrimitive("Boolean", true, DATA_FACTORY.createPrimitiveBoolean))
+        primTypes.add(createPrimitive("Date", true, DATA_FACTORY.createPrimitiveDate))
+        primTypes.add(createPrimitive("DateTime", false, DATA_FACTORY.createPrimitiveString))
+        primTypes.add(createPrimitive("Password", false, DATA_FACTORY.createPrimitiveString))
+        primTypes.add(createPrimitive("Unspecified", true, DATA_FACTORY.createPrimitiveUnspecified))
+        // Because LEMMA needs default values short and char are added.
+        // However, OpenAPI v3.0.3 does not specifically describe these.
+        primTypes.add(createPrimitive("Short", true, DATA_FACTORY.createPrimitiveShort))
+        primTypes.add(createPrimitive("Char", true, DATA_FACTORY.createPrimitiveCharacter))
+
+        technology.primitiveTypes.addAll(primTypes)
+    }
+
+    def TechnologySpecificPrimitiveType createPrimitive(
+        String name,
```

Comment:
If `name` and `default` fit into one line, pull them up on the same line as the method definition.